### ISSUE TITLE
Is "Lipro Balance Charger" supported?

### ISSUE THREAD

**Issue by Harvie** [stawel/cheali-charger#214]
Can i use this with cheap chinese "Lipro Balance Charger" (fake imax b6)?
I can't see any label on chip, but it's hidden under lcd, so it's hard to see.
If so can you please add it to hardware list in readme?

<img src="https://statics3.seeedstudio.com/product/LiProCharger_02.jpg" />

**Comment by stawel** [comment]
It's hard to say, they all look the same from outside, try to lift a bit your LCD and check your CPU
, or at least send a picture of your PCB.
If there is no label on your CPU then it's probably not supported.

**Comment by Harvie** [comment]
I've made some photos and even managed to lift the lcd a bit... Can you tell more this way?

![dsc_0108](https://cloud.githubusercontent.com/assets/276504/24083251/06d5d8a4-0cd4-11e7-9427-1b5c4b6b552a.JPG)
![dsc_0109](https://cloud.githubusercontent.com/assets/276504/24083253/06d7c52e-0cd4-11e7-856a-1dc0330d5ffc.JPG)
![dsc_0110](https://cloud.githubusercontent.com/assets/276504/24083256/06d90efc-0cd4-11e7-8d76-ded45ccd336f.JPG)
![dsc_0111](https://cloud.githubusercontent.com/assets/276504/24083252/06d7882a-0cd4-11e7-8814-a52299fb5e46.JPG)
![dsc_0112](https://cloud.githubusercontent.com/assets/276504/24083255/06d86984-0cd4-11e7-9906-a11949834730.JPG)
![dsc_0113](https://cloud.githubusercontent.com/assets/276504/24083254/06d824d8-0cd4-11e7-8219-3976ae51f488.JPG)
![dsc_0114](https://cloud.githubusercontent.com/assets/276504/24083257/06eaf81a-0cd4-11e7-8472-db273b7b969b.JPG)
![dsc_0115](https://cloud.githubusercontent.com/assets/276504/24083258/06f002ce-0cd4-11e7-827c-326ff4e41593.JPG)

**Comment by Harvie** [comment]
I've done little googling. And it might, just might be a **z51f3220**. I am not sure... Any luck with this MCU?

**Comment by stawel** [comment]
Yes, it may be a z51f3220, unfortunately this CPU will not be supported,
I've tried to add it to the project but it's architektur is just to old.

**Comment by Harvie** [comment]
It seems there are people on internet having same problem... I guess there's no lightweight cheali alternative targeting these devices yet?

**Comment by stawel** [comment]
I'm not aware of any.

**Comment by intruder777** [comment]
I have similar device (marked as 80W). Processor is labeled, but the label is scratched out with laser. However the first line can be clearly read as "ABOV". And people on the net say that second line is "MC96F6432".
Here it is:
http://abov.co.kr/en/index.php?Depth1=3&Depth2=1&Depth3=1&Depth4=1&Item=MC96F6432
http://aitendo3.sakura.ne.jp/aitendo_data/product_img/microcontroller/MC96F6432/MC96F6432_V1.2.pdf

Is it possible to port cheali-carger to this processor?

**Comment by diggit** [comment]
Hi @intruder777,
This MCU is 8051 based and these are not supported by cheali. AFAIK there is no free multiplatform c++ compiler for 8051. Another problem could be only 1kiB ram. (atmega32 has 2kiB)
More here https://github.com/stawel/cheali-charger/issues/106#issuecomment-222520952
